Move the edit button in the site editor sidebar to a contextual widget

youknowriad opened the pull request:
Related #36667

## What?

This PR redesigns the sidebar of the browse mode to match the video shared here https://user-images.githubusercontent.com/846565/208665227-82537dfb-2893-4fb6-8f84-823f1e70cc07.mp4

## Why?

We have noticed that the "edit" button is a bit hidden and lacks context/connection to the frame. This PR adds a "widget" on top of the sidebar to add that missing context and move the edit button there for more clarity.

## How?

 - In order to so, I had to redesign the whole layout with a new structure. No more grid, now, it's all flex box
 - The header is fixed at the top
 - The island is fixed on top of the sidebar, that way it can morph into the header when moving to edit mode.

**Note**

 - At the moment, I didn't design the "title" yet, the piece of UI that show be showing the active template or template part title loaded in the frame. I'll follow-up with that quickly. I focused on the animation, positions...
 - In this initial PR, I'm not going to be adding any shortcuts to the header of the "edit" mode.

https://user-images.githubusercontent.com/272444/208882700-9cc76cb1-d1e2-4b72-915a-a2c1fdb6751a.mov

youknowriad committed:
Move the edit button in the site editor sidebar to a contextual widget

youknowriad committed:
Fix sidebar scrolling

youknowriad committed:
Do not hide the inserter in the header

youknowriad committed:
Small fix

jasmussen commented:
Thanks for this. I think there's a lot of sense in isolating the site information in a "hub" like that, letting the waterfall navigation slide below it. It is also a way, potentially, to allow for more context to show next to the Edit button in the future.

Can the edit button be 32px tall?

jameskoster commented:
> Can the edit button be 32px tall?

Also the W logo / site icon for balance?

The W logo is currently 30x30, inside a 36px button, so will need some tweaking.

Edit: I noticed the sidebar seems to have shrunk a bit (width), and the top toolbar animation has disappeared.

Edit 2: One other small thing: When moving from browse to edit, the nav panel slides out to the left. Instead of sliding back in when moving from edit to browse it appears instantly. I don't have a strong preference re sliding vs fading, but they should probably be consistent?

jameskoster committed:
Update hub shadow

youknowriad commented:
> One other small thing: When moving from browse to edit, the nav panel slides out to the left. Instead of sliding back in when moving from edit to browse it appears instantly. I don't have a strong preference re sliding vs fading, but they should probably be consistent?

Unfortunately, this animation is controlled by a lower-level reusable component "Navigator", a fix might not be easy here. Maybe I can try to disable the sliding and only leave fading.

youknowriad committed:
Resize the edit button

youknowriad committed:
Restore the sidebar width

youknowriad committed:
Restore the top bar animation

youknowriad commented:
- I fixed the edit button height
- I fixed the sidebar width
- I fixed the top bar animation
-  Didn't manage to fix the exit animation of the sidebar content

How is this looking?

jasmussen commented:
I think it's looking good!

![anim](https://user-images.githubusercontent.com/1204802/209097541-bfb26cd3-b624-4d64-8792-a8a88accec92.gif)

At the risk of repeating Jay on animation, the inserter button appearing "under" the hub animation feels awkward. Could we speed up the hub animation so it finishes before the white toolbar drops down? Or can we animate width of the white bar (might be awkward).

Conceptually, it makes sense that the hub is an element that floats _above_ the bottom dark gray canvas, and _above_ the white toolbar and inspector elements in elevation. That's already the case in this animation. But I think either speeding up the hub-collapsing animation might thread the needle. Alternately, maybe the contents of the top toolbar (all the buttons) could fade in as the toolbar drops down. That might also give the appearance that they aren't interactive yet.

youknowriad committed:
Speeding up the hub animation

youknowriad commented:
The hub animates faster now.

jorgefilipecosta commented on packages/edit-site/src/components/layout/style.scss:
```diff
@@ -1,80 +1,102 @@
+	left: $canvas-padding;
+	width: calc(100vw - #{$canvas-padding * 2});
+	height: $hub-height;
+	z-index: 3;
```
Should we remove this hardcoded value and create a z-index function value for this?

jorgefilipecosta commented on packages/edit-site/src/components/layout/style.scss:
```diff
@@ -1,80 +1,102 @@
 	height: $header-height;
 	display: flex;
-	justify-content: space-between;
+	z-index: 2;
```
Same comment as above.

jorgefilipecosta reviewed (commented):
Some end-to-end test failures may be relevant.

youknowriad committed:
Fix z-index and e2e tests

jasmussen reviewed (approved):
This feels like a solid upgrade over trunk, especially if paired with #46736, so greenlighting it as such:

![status](https://user-images.githubusercontent.com/1204802/209349378-dcc6011b-860a-456b-8638-13aa14325140.gif)

I still think we need to make the nav sidebar 360px wide, and I wish there was a way we could gate the browse view mostly as an experiment, but since that's not really feasible. at least his increases contextuality and usability a fair bit.

jorgefilipecosta reviewed (approved):
Worked well on my tests 👍 And the code looks good.
